vignettes/datatable-joins.Rmd

```diff
@@ -0,0 +1,687 @@
+         on = .(id = product_id)]
+```
+
+#### 3.1.2. Alternatives to define the `on` argument
+
+In all the prior example we have pass the column names we want to match to the `on` argument but `data.table` also have alternatives to that syntax.
+
+- **Natural join**: Selects the columns to perform the match based on common column names. To illustrate this method, let's change the column of `Products` table from `id` to `product_id` and use the keyword `.NATURAL`.
+
+```{r}
+ProductsChangedName = setnames(copy(Products), "id", "product_id")
+ProductsChangedName
+
+ProductsChangedName[ProductReceived, on = .NATURAL]
+```
+
+- **Keyed join**: Selects the columns to perform the match based on keyed columns regardless of their names.To illustrate this method, we need to define keys in the same order for both tables.
+
+```{r}
+ProductsKeyed = setkey(copy(Products), id)
+key(ProductsKeyed)
+
+ProductReceivedKeyed = setkey(copy(ProductReceived), product_id)
+key(ProductReceivedKeyed)
+
+ProductsKeyed[ProductReceivedKeyed]
+```
+
+#### 3.1.3. Operations after joining
+
+Most of the time after a join is complete we need to make some additional transformations. To make so we have the following alternatives:
+
+- Chaining a new instruction by adding a pair of brakes `[]`.
+- Passing a list with the columns that we want to keep or create to the `j` argument.
+
+Our recommendation is to use the second alternative if possible, as it is **faster** and uses **less memory** than the first one.
+
+
+##### 3.1.3.1. Managing shared column Names with the j argument
+
+The `j` argument has great alternatives to manage joins with tables **sharing the same names for several columns**. By default all columns are taking their source from the the `x` table, but we can also use the `x.` prefix to make clear the source and use the prefix `i.` to use any column form the table declared in the `i` argument of the `x` table.
+
+Going back to the little supermarket, after updating the `ProductReceived` table with the `Products` table, it seems convenient apply the following changes:
+
+- Changing the columns names from `id` to `product_id` and from `i.id` to `received_id`.
+- Adding the `total_value`.
+
+```{r}
+Products[ProductReceived,
+         on = c("id" = "product_id"),
+         j = .(product_id = x.id,
+               name = x.name,
+               price,
+               received_id = i.id,
+               date = i.date,
+               count,
+               total_value = price * count)]
+```
+
+
+##### 3.1.3.2. Summarizing with on in data.table
+
+We can also use this alternative to return aggregated results based columns present in the `x` table.
+
+For example, we might interested in how much money we expend buying products each date regardless the products.
+
+```{r}
+dt1 =
+  ProductReceived[Products,
+                  on = c("product_id" = "id"),
+                  by = .EACHI,
+                  j = .(total_value_received  = sum(price * count))]
```

Comment:
please consider alternative whitespace formatting below.
when each argument of square brackets is on its own line, each argument can be commented 
```r
dt1 = ProductReceived[
  Products,
  on = c("product_id" = "id"),
  by = .EACHI,
  j = .(total_value_received  = sum(price * count))
]

Reply:
I like this format.